lib/src/Base/Optim/Cobyla.cxx

```diff
@@ -216,7 +223,21 @@ int Cobyla::ComputeObjectiveAndConstraint(int n,
       if (!SpecFunc::IsNormal(inP[i]))
         throw InvalidArgumentException(HERE) << "Cobyla got a nan input value";
```

Comment:
Based on the error message, shouldn't we check for nans instead of non-normals?

Reply:
inf too